pallets/runtime/src/general_tm.rs

```diff
@@ -91,10 +91,21 @@ pub struct AssetRule {
     pub receiver_rules: Vec<RuleData>,
 }
 
+/// Details of a rule
+#[derive(codec::Encode, codec::Decode, Default, Clone, PartialEq, Eq, Debug)]
+pub struct RuleDetails {
+    /// Asset rule
+    pub asset_rule: AssetRule,
+
+    /// Unique identifier of the asset rule
+    pub rule_id: u32,
```

Comment:
I think `rule_id` should be part of `AssetRule`. I don't see the need for another wrapper struct here.

Reply:
Yeah, we don't need it. I intentionally added this way to maintain the data structure to pass into functions like `add_active_rule`, `change_asset_rule`.  Otherwise, we have to pass 2 different variables as sender_rules & reciever_rule..

Reply:
Both sender_rules & reciever_rule should have a rule_id. 
As far as passing structs in functions go, we can create a new struct for that without rule_id.

That way, our storage will be a little cleaner and faster to serialize and deserialize.